index.js

```diff
@@ -0,0 +1,214 @@
+import * as Sentry from '@sentry/node'
+import getRawBody from 'raw-body'
+import httpAssert from 'http-assert'
+import * as ethers from 'ethers'
+import { json, status } from 'http-responders'
+import assert from 'node:assert'
+
+const maxScore = BigInt(1e15)
+// https://github.com/filecoin-station/spark-impact-evaluator/blob/fd64313a96957fcb3d5fda0d334245601676bb73/test/Spark.t.sol#L11C39-L11C65
+const roundReward = 456621004566210048n
+
+const handler = async (req, res, redis, signerAddresses) => {
+  if (req.method === 'POST' && req.url === '/scores') {
+    await handleIncreaseScores(req, res, redis, signerAddresses)
+  } else if (req.method === 'POST' && req.url === '/paid') {
+    await handlePaidScheduledRewards(req, res, redis, signerAddresses)
+  } else if (req.method === 'GET' && req.url === '/scheduled-rewards') {
+    await handleGetAllScheduledRewards(res, redis)
+  } else if (req.method === 'GET' && req.url.startsWith('/scheduled-rewards/')) {
+    await handleGetSingleScheduledRewards(req, res, redis)
+  } else if (req.method === 'GET' && req.url === '/log') {
+    await handleGetLog(res, redis)
+  } else {
+    status(res, 404)
+  }
+}
+
+const validateSignature = (signature, participants, signerAddresses) => {
+  httpAssert(
+    typeof signature === 'object' && signature !== null,
+    400,
+    '.signature should be an object'
+  )
+  httpAssert.deepEqual(
+    Object.keys(signature).sort(),
+    ['r', 's', 'v'],
+    400,
+    '.signature should have keys .r, .s and .v'
+  )
+
+  const digest = ethers.solidityPackedKeccak256(
+    ['address[]', 'int256[]'],
+    [Object.keys(participants), Object.values(participants)]
```

Comment:
This assumes that `Object.keys(JSON.parse(text))` preserves the order of the keys. I think this assumption is true for V8/Node.js, so we should be fine here. Other runtimes, like Go or Java, don't provide this guarantee, though.

Reply:
Yeah, relying on the order is hacky, but works for us here

Reply:
Solved!